numpy/_core/src/common/simd/lsx/operators.h

```diff
@@ -0,0 +1,301 @@
+#define npyv_cmpgt_u64(A, B) __lsx_vslt_du(B, A)
+
+// unsigned greater than or equal
+#define npyv_cmpge_u8(A, B)  __lsx_vsle_bu(B, A)
+#define npyv_cmpge_u16(A, B) __lsx_vsle_hu(B, A)
+#define npyv_cmpge_u32(A, B) __lsx_vsle_wu(B, A)
+#define npyv_cmpge_u64(A, B) __lsx_vsle_du(B, A)
+
+// less than
+#define npyv_cmplt_u8  __lsx_vslt_bu
+#define npyv_cmplt_s8  __lsx_vslt_b
+#define npyv_cmplt_u16 __lsx_vslt_hu
+#define npyv_cmplt_s16 __lsx_vslt_h
+#define npyv_cmplt_u32 __lsx_vslt_wu
+#define npyv_cmplt_s32 __lsx_vslt_w
+#define npyv_cmplt_u64 __lsx_vslt_du
+#define npyv_cmplt_s64 __lsx_vslt_d
+
+// less than or equal
+#define npyv_cmple_u8  __lsx_vsle_bu
+#define npyv_cmple_s8  __lsx_vsle_b
+#define npyv_cmple_u16 __lsx_vsle_hu
+#define npyv_cmple_s16 __lsx_vsle_h
+#define npyv_cmple_u32 __lsx_vsle_wu
+#define npyv_cmple_s32 __lsx_vsle_w
+#define npyv_cmple_u64 __lsx_vsle_du
+#define npyv_cmple_s64 __lsx_vsle_d
+
+// precision comparison
+#define npyv_cmpeq_f32  __lsx_vfcmp_ceq_s
+#define npyv_cmpeq_f64  __lsx_vfcmp_ceq_d
+#define npyv_cmpneq_f32 __lsx_vfcmp_cune_s
+#define npyv_cmpneq_f64 __lsx_vfcmp_cune_d
+#define npyv_cmplt_f32  __lsx_vfcmp_clt_s
+#define npyv_cmplt_f64  __lsx_vfcmp_clt_d
+#define npyv_cmple_f32  __lsx_vfcmp_cle_s
+#define npyv_cmple_f64  __lsx_vfcmp_cle_d
+#define npyv_cmpgt_f32(A, B) npyv_cmplt_f32(B, A)
+#define npyv_cmpgt_f64(A, B) npyv_cmplt_f64(B, A)
+#define npyv_cmpge_f32(A, B) npyv_cmple_f32(B, A)
+#define npyv_cmpge_f64(A, B) npyv_cmple_f64(B, A)
+
+// check special cases
+NPY_FINLINE npyv_b32 npyv_notnan_f32(npyv_f32 a)
+{ return __lsx_vfcmp_cor_s(a, a); }    //!nan,return:ffffffff
+NPY_FINLINE npyv_b64 npyv_notnan_f64(npyv_f64 a)
+{ return __lsx_vfcmp_cor_d(a, a); }
+
+// Test cross all vector lanes
+// any: returns true if any of the elements is not equal to zero
+// all: returns true if all elements are not equal to zero
+#define NPYV_IMPL_LSX_ANYALL(SFX)                   \
+    NPY_FINLINE bool npyv_any_##SFX(npyv_##SFX a)   \
+    { return __lsx_vmsknz_b((__m128i)a)[0] != 0; }  \
+    NPY_FINLINE bool npyv_all_##SFX(npyv_##SFX a)   \
+    { return __lsx_vmsknz_b((__m128i)a)[0] == 0xffff; }
+NPYV_IMPL_LSX_ANYALL(b8)
+NPYV_IMPL_LSX_ANYALL(b16)
+NPYV_IMPL_LSX_ANYALL(b32)
+NPYV_IMPL_LSX_ANYALL(b64)
+#undef NPYV_IMPL_LSX_ANYALL
+
+#define NPYV_IMPL_LSX_ANYALL(SFX, TSFX, MASK)       \
+    NPY_FINLINE bool npyv_any_##SFX(npyv_##SFX a)   \
+    {                                               \
+        return __lsx_vmsknz_b(                      \
+            __lsx_vseq_##TSFX(a, npyv_zero_##SFX()) \
+        )[0] != MASK;                               \
+    }                                               \
+    NPY_FINLINE bool npyv_all_##SFX(npyv_##SFX a)   \
+    {                                               \
+        return __lsx_vmsknz_b(                      \
+            __lsx_vseq_##TSFX(a, npyv_zero_##SFX()) \
+        )[0] == 0;                                  \
```

Comment:
use __lsx_vmsknz_b(a)[0] == 0xffff

Reply:
All not equal to zero functions cannot be used directly, vmsknz_b just suit byte type, it will failed when use other data type.